src/sage/graphs/generic_graph.py

```diff
@@ -938,6 +939,143 @@ def _latex_(self):
+
+        ::
+
+            sage: tikz = g.tikz(format='tkz_graph')
+            sage: _ = tikz.pdf()      # not tested
+
+        Using another value for ``prog``::
+
+            sage: tikz = g.tikz(prog='neato',        # long time (3s), optional - dot2tex graphviz
+            ....:               color_by_label=True)
+            sage: _ = tikz.pdf()      # not tested
+
+        Using another value for ``rankdir``::
+
+            sage: tikz = g.tikz(rankdir='right')     # long time (3s), optional - dot2tex graphviz
+            sage: _ = tikz.pdf()      # not tested
+
+        Using subgraphs clusters (broken when using labels, see
+        :issue:`22070`)::
+
+            sage: S = FiniteSetMaps(5)
+            sage: I = S((0,1,2,3,4))
+            sage: a = S((0,1,3,0,0))
+            sage: b = S((0,2,4,1,0))
+            sage: roots = [I]
+            sage: succ = lambda v: [v*a,v*b,a*v,b*v]
+            sage: R = RecursivelyEnumeratedSet(roots, succ)
+            sage: G = R.to_digraph()
+            sage: G
+            Looped multi-digraph on 27 vertices
+            sage: C = G.strongly_connected_components()
+            sage: tikz = G.tikz(subgraph_clusters=C)    # optional - dot2tex graphviz
+            sage: _ = tikz.pdf()      # not tested
+
+        An example coming from ``graphviz_string`` documentation in SageMath::
+
+            sage: # needs sage.symbolic
+            sage: f(x) = -1 / x
+            sage: g(x) = 1 / (x + 1)
+            sage: G = DiGraph()
+            sage: G.add_edges((i, f(i), f) for i in (1, 2, 1/2, 1/4))
+            sage: G.add_edges((i, g(i), g) for i in (1, 2, 1/2, 1/4))
+            sage: tikz = G.tikz(format='dot2tex')    # optional - dot2tex graphviz
+            sage: _ = tikz.pdf()                       # not tested
+            sage: def edge_options(data):
+            ....:     u, v, label = data
+            ....:     options = {"color": {f: "red", g: "blue"}[label]}
+            ....:     if (u,v) == (1/2, -2): options["label"]       = "coucou"; options["label_style"] = "string"
+            ....:     if (u,v) == (1/2,2/3): options["dot"]         = "x=1,y=2"
+            ....:     if (u,v) == (1,   -1): options["label_style"] = "latex"
+            ....:     if (u,v) == (1,  1/2): options["dir"]         = "back"
+            ....:     return options
+            sage: tikz = G.tikz(format='dot2tex',   # optional - dot2tex graphviz
+            ....:               edge_options=edge_options)
+            sage: _ = tikz.pdf()      # not tested
+
+        """
+        # use format dot2tex by default
+        if format is None:
+            from sage.features import PythonModule
+            if PythonModule("dot2tex").is_present():
+                format = 'dot2tex'
+            else:
+                format = 'tkz_graph'
+
+        # by default draw edge_labels only for dot2tex (because tkz_graph
+        # puts None everywhere which is ugly)
+        if edge_labels is None:
```

Comment:
at this point, you have only tested if `format is None`. Why not testing if format is `'dot2tex'` or `'tkz_graph'` before ?

Reply:
In fact, I prefer not to test for values of format, because this is done in the `set_options` and `set_option` methods inside `graph_latex.py` file. If one day, someone defines a new format. Then, it is better if the `tikz` method is independent of it and just works for the new format. There will be no default values for it, but at least it will work.

Reply:
I made a commit where I remove the ValueError raised.